File: Programmers/Level2/스킬트리.py
def check(skill, spell):
    for i in range(len(skill)):
        if i < len(spell):
            if skill[i] != spell[i]:
                return False
    return True

def solution(skill, skill_trees):
    answer = 0
    for spell in skill_trees:
        for s in spell:
            if s not in skill:
                spell = spell.replace(s, '')
        if check(skill, spell):
            answer += 1
                
    return answer

skill = 'CBD'
skill_trees = 	["BACDE", "CBADF", "AECB", "BDA"]

print(solution(skill, skill_trees))

File: Programmers/Level2/스킬트리/스킬트리.js
function solution(skill, skill_trees) {
  const order = skill.split("");
  let count = 0;
  for (let i = 0; i < skill_trees.length; i++) {
    const mySkill = skill_trees[i]
      .split("")
      .filter((ele) => order.includes(ele))
      .join("");
    if (mySkill === skill.substring(0, mySkill.length)) {
      count++;
    }
  }
  return count;
}

const skill = "CBD";
const skill_trees = ["BACDE", "CBADF", "AECB", "BDA"];
console.log(solution(skill, skill_trees));
